### ISSUE TITLE
java.lang.NoSuchMethodError thrown with the latest Ammonite

### ISSUE THREAD

**Issue by DoruM** [scalameta/scalameta#2180]
**Describe the bug**
When I use Ammonite, the example code from https://scalameta.org/docs/trees/guide.html throws a NoSuchMethodError
.
```
 java.lang.NoSuchMethodError: scala.meta.Dialect.allowSignificantIndentation()Z
   scala.meta.internal.parsers.ScalametaParser.canBeFollowedByColonEol(ScalametaParser.scala:240)
   scala.meta.internal.parsers.ScalametaParser.loop$1(ScalametaParser.scala:317)
   scala.meta.internal.parsers.ScalametaParser.x$40$lzycompute(ScalametaParser.scala:505)
   scala.meta.internal.parsers.ScalametaParser.x$40(ScalametaParser.scala:272)
   scala.meta.internal.parsers.ScalametaParser.parserTokens$lzycompute(ScalametaParser.scala:272)
   scala.meta.internal.parsers.ScalametaParser.parserTokens(ScalametaParser.scala:272)
   scala.meta.internal.parsers.ScalametaParser$SimpleTokenIterator.<init>(ScalametaParser.scala:552)
   scala.meta.internal.parsers.ScalametaParser.<init>(ScalametaParser.scala:550)
   scala.meta.internal.parsers.ScalametaParser$$anon$245.apply(ScalametaParser.scala:4762)
   scala.meta.parsers.Api$XtensionParseDialectInput.parse(Api.scala:25)
   scala.meta.parsers.Api$XtensionParseInputLike.parse(Api.scala:14)
   ammonite.$sess.cmd2$.<clinit>(cmd2.sc:1)
```

**To reproduce**
1. Install Ammonite with the command:
```sh
sudo sh -c '(echo "#!/usr/bin/env sh" && curl -L https://github.com/lihaoyi/Ammonite/releases/download/2.3.8/2.13-2.3.8) > /usr/local/bin/amm && chmod +x /usr/local/bin/amm' && amm
```
2. Run the following commands:
```scala
import $ivy.`org.scalameta::scalameta:4.4.0`, scala.meta._
val program = """object Main extends App { print("Hello!") }""" 
val tree = program.parse[Source].get
```
3. Observe the stacktrace described above

**Expected behaviour**

The output of the parsing should be shown: 
```scala
tree: Source = Source(List(Defn.Object(List(), Term.Name("Main"), Template(List(), List(Init(Type.Name("App"), , List())), Self(, None), List(Term.Apply(Term.Name("print"), List(Lit.String("Hello!"))))))))
```

**Additional context**
This doesn't reproduce with version 4.3.20.
Versions 4.3.21-4.3.22 also throw NoSuchMethodError but it is in a different place 

```
java.lang.NoSuchMethodError: scala.meta.Dialect.allowEndMarker()Z
  scala.meta.internal.parsers.ScalametaParser$EndMarkerIntro$.unapply(ScalametaParser.scala:607)     
```
System: Ubuntu 20
Shell: zsh
JVM: 8.0.275.open-adpt (AdoptOpenJDK)

**Comment by olafurpg** [comment]
Thank you for reporting! This is an issue that can be fixed by upgrading Scalameta to 4.4.0 in the Ammonite build here

https://github.com/lihaoyi/Ammonite/blob/d5beecc87a21fe9f559b34ef64aac1d194572221/build.sc#L215

A workaround is to use Scalameta 4.3.20 instead of 4.4.0.

I'm not sure there's much we can do on the Scalameta side to fix this issue. The 4.4.0 release is backwards binary compatible with 4.3.20 but it's not forwards compatible. The root cause of this problem is that Ammonite includes an older version of Scalameta on the root classloader in the Ammonite REPL. A permanent fix would be to layer the classloaders in the Ammonite REPL somehow so that users can optionally bring in their own version of Scalameta that doesn't conflict with the Scalameta version in the Ammonite build.

**Comment by tgodzik** [comment]
Please use `amm --thin` when working with scalameta. That will make sure that the dependency is not on the classpath.
